tests/test_solvers/test_harbor_solver.py

```
@@ -226,6 +226,41 @@ async def test_runner_patch_disables_visualizer(self):
+
+        timeout_patch = next(
+            (s for s in decoded_scripts if "llm_timeout" in s and "LLM_TIMEOUT" in s),
+            None,
+        )
+        assert timeout_patch is not None, "LLM timeout patch script not emitted"
+
+        timeout_patch = timeout_patch.replace(
+            "p = '/installed-agent/run_agent.py'",
+            f"p = {str(runner)!r}",
+        )
+        exec(compile(timeout_patch, "llm_timeout_patch.py", "exec"), {})
```

Ruff (0.15.17)

[error] Use of `exec` detected

(S102)